contracts/zero-ex/contracts/src/external/TransformerDeployer.sol

```diff
@@ -0,0 +1,226 @@
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+
+/// @dev A contract with a `die()` function.
+interface IKillable {
+    function die() external;
+}
+
+/// @dev Shared deployer contract for ERC20 transformers implementing some
+///      basic membership governance.
+///      All members can create or kill transformers individually.
+///      A majority vote is required to add or remove memebers.
+contract TransformerDeployer {
+
+    /// @dev Emitted when a contract is deployed via `deploy()`.
+    event Deployed(address deployedAddress, address member);
+    /// @dev Emitted when a contract is killed via `kill()`.
+    event Kill(address target, address member);
+
+    /// @dev Represents a member (de)registration vote.
+    ///      Should be passed into `addMembers()` and `removeMembers()` with
+    ///      accompanying signature.
+    struct Vote {
+        // Members to add or remove.
+        address[] members;
+        // How long this vote is valid for.
+        uint256 expirationTime;
+    }
+
+    /// @dev The EIP712 typehash for `addMembers()` votes.
+    bytes32 private immutable ADD_MEMBERS_VOTE_TYPEHASH = keccak256(
+        "AddMembersVote(address[] member,uint256 nonce,bytes signature)"
+    );
+    /// @dev The EIP712 typehash for `removeMembers()` votes.
+    bytes32 private immutable REMOVE_MEMBERS_VOTE_TYPEHASH = keccak256(
+        "RemoveMembersVote(address[] member,uint256 nonce,bytes signature)"
+    );
+
+    /// @dev The EIP712 domain separator.
+    bytes32 public immutable EIP712_DOMAIN_SEPARATOR;
+    /// @dev The current deployment nonce of this contract.
+    ///      Unlike EOAs, contracts start with a nonce of 1.
+    uint256 public deploymentNonce = 1;
+    /// @dev The number of registered members.
+    uint256 public memberCount;
+    /// @dev Whether an address is a member.
+    mapping(address => bool) public isMember;
+    /// @dev Whether a vote was consumed.
+    mapping(bytes32 => bool) public isVoteConsumed;
+
+    /// @dev Only a valid member can call the function.
+    modifier onlyMember() {
+        require(isMember[msg.sender], "TransformerDeployer/ONLY_CALLABLE_By_MEMBER");
+        _;
+    }
+
+    /// @dev Create this contract and seed the initial members.
+    constructor(address[] memory members) public {
+        uint256 chainId;
+        assembly { chainId := chainid() }
+        EIP712_DOMAIN_SEPARATOR = keccak256(abi.encode(
+            keccak256(
+                "EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)"
+            ),
+            keccak256(bytes('TransformerDeployer')),
+            keccak256(bytes('1.0.0')),
+            chainId,
+            address(this)
+        ));
+        _addMembers(members);
+    }
+
+    /// @dev Registers new members. Must attach majority votes.
+    function addMembers(Vote[] memory votes, bytes[] memory signatures)
```

Comment:
I'd prefer that we change the `Vote` struct to:
```
    struct Vote {
        uint256 expirationTime;
        bytes signature;
    }
```
and the function signature to:
```suggestion
    function addMembers(address[] memory members, Vote[] memory votes)
```
so that we don't need to check that the `members` hash is the same for all the votes

Reply:
moot now, since we're now just going with  a regular authorizable pattern